File: StackWithArrays/stack.py
from sys import maxsize

def createStack():
  stack = []
  return stack

def isEmpty(stack):
  return len(stack) == 0

def push(stack, item):
  stack.append(item)

def pop(stack):
  if (isEmpty(stack)):
    return str(-maxsize -1)
  return stack.pop()

def peek(stack):
  if (isEmpty(stack)):
    return str(-maxsize -1)
  return stack[len(stack) - 1]


if __name__ == '__main__':
  stack = createStack()
  push(stack, str(1))
  push(stack, str(2))
  push(stack, str(3))
  pop(stack)

File: StackWithArrays/stack.java
class Stack {
  static final int MAX = 10;
  int top;
  int a[] = new int[MAX];

  Stack() { top = -1; }

  boolean push(int x) {
    if (top >= (MAX - 1))
      return false;
    a[++top] = x;
    return true;
  }

  int pop() {
    if (top < 0)
      return 0;
    int x = a[top--];
    return x;
  }

  int peek() {
    if (top < 0)
      return 0;
    int x = a[top];
    return x;
  }

  boolean isEmpty() {
    return (top < 0);
  }
}

class Main {
  public static void main(String args[]) {
    Stack s = new Stack();
    s.push(1);
    s.push(2);
    s.push(3);
    s.pop();
  }
}
